### PR TITLE
Update porting guide

### PR DESCRIPTION
**Pull request by abadger**
##### SUMMARY
Porting guide for Asnible-4

##### ISSUE TYPE

- Docs Pull Request

##### ADDITIONAL INFORMATION
@felixfontein @samccann @acozine I noticed a few things in the generated file that should be changed.  I modified them in a way I thought would be okay in the second commit: e4650a58c1cdcc9d1face38a73b8208af45a346d  If you all think that the updated wording is good, I can get those into a PR for `antsibull{-changelog} ` as well.

### CODE REVIEW

File: docs/docsite/rst/porting_guides/porting_guide_4.rst [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+=======================
+Ansible 4 Porting Guide
+=======================
+
+.. contents::
+  :local:
+  :depth: 2
+
+
+If you are upgrading from Ansible 2, please first consult the Ansible 3 porting guide before continuing with the Ansible 4 porting guide.
```
Comment (samccann) [inline docs/docsite/rst/porting_guides/porting_guide_4.rst]:
```suggestion
If you are upgrading from Ansible 2.x, please first consult the Ansible 3 porting guide before continuing with the Ansible 4 porting guide.
```

Comment (samccann) [reply docs/docsite/rst/porting_guides/porting_guide_4.rst]: up to you whether you accept that change. Reading 'Ansible 2' just felt weird to me.

**Review by samccann — COMMENTED** [review]
just a nit

File: docs/docsite/rst/porting_guides/porting_guide_4.rst (hunk 1 above)
Comment (felixfontein) [reply docs/docsite/rst/porting_guides/porting_guide_4.rst]: Why not being explicit here and mention 2.10? If upgrading from 2.9, also the 2.10 porting guide should also be read, and so on for versions before that.

**Comment by abadger** [comment]
I think I've fixed the `ansible-test build-docs` error but in doing so, I discovered one wrinkle in our split docsite.  The present build-docs test is only testing `make coredocs`.  If we want to test everything we probably need to run the build-docs test twice.  Once to `make coredocs` and once to `make webdocs`.

**Comment by abadger** [comment]
New version built from updated antsibull sources will arrive shortly.

**Comment by abadger** [comment]
Okay, porting guide updated with what the latest antsibull generates.  It now contains the porting guide entries from ansible-core-2.11  (Thanks felixfontein!)

File: docs/docsite/sphinx_conf/core_conf.py [hunk 2]
```diff
@@ -99,7 +99,8 @@
     'scenario_guides',
     'porting_guides/porting_guides.rst',
     'porting_guides/porting_guide_2*',
-    'porting_guides/porting_guide_3*',
+    'porting_guides/porting_guide_3.rst',
+    'porting_guides/porting_guide_4.rst*',
```
Comment (acozine) [inline docs/docsite/sphinx_conf/core_conf.py]:
```suggestion
    'porting_guides/porting_guide_4.rst',
```

**Review by acozine — COMMENTED** [review]
Overall looks good. Just one small tweak to the core conf.

**Comment by abadger** [comment]
rebuild_merge

**Pull request merged**